josh-mysql/src/error.js

```diff
@@ -0,0 +1,12 @@
+class CustomError extends Error {
```

Comment:
Not specifically related to the PR, but this file. Wouldn't it be better to have a core folder in order to avoid duplicating classes/utilities that are used over multiple providers?
cc @eslachance

Reply:
Go right ahead and PR that in, darling! ^_^